crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -40,22 +40,10 @@ pub struct CheckpointObjectChanges {
 }
 
 #[async_trait]
-impl Worker for ObjectsSnapshotProcessor {
+impl Worker for ObjectsSnapshotHandler {
+    // ?? change Worker trait to use Arc<CheckpointData> to avoid clone
```

Comment:
@phoenix-o wdyt of changing Worker trait to use Arc<CheckpointData> to avoid checkpoint data clone here?

Reply:
I'd argue that this particular handler should be reworked so that we don't need to send the whole checkpoint off. The parsing of data out of the checkpoint should be done here vs where ever the receiving part of the channel lives